src/ddb/ddb_cmd_options.c

```diff
@@ -299,6 +515,57 @@ ddb_parse_cmd_args(struct ddb_ctx *ctx, struct argv_parsed *parsed, struct ddb_c
 		info->dci_cmd = DDB_CMD_LOAD;
 		return load_option_parse(ctx, &info->dci_cmd_option.dci_load, parsed);
 	}
+	if (same(cmd, COMMAND_NAME_DUMP_ILOG)) {
+		info->dci_cmd = DDB_CMD_DUMP_ILOG;
+		return dump_ilog_option_parse(ctx, &info->dci_cmd_option.dci_dump_ilog, parsed);
+	}
+	if (same(cmd, COMMAND_NAME_PROCESS_ILOG)) {
+		info->dci_cmd = DDB_CMD_PROCESS_ILOG;
+		return process_ilog_option_parse(ctx, &info->dci_cmd_option.dci_process_ilog,
+						 parsed);
+	}
+	if (same(cmd, COMMAND_NAME_RM_ILOG)) {
+		info->dci_cmd = DDB_CMD_RM_ILOG;
+		return rm_ilog_option_parse(ctx, &info->dci_cmd_option.dci_rm_ilog, parsed);
+	}
+	if (same(cmd, COMMAND_NAME_DUMP_DTX)) {
+		info->dci_cmd = DDB_CMD_DUMP_DTX;
+		return dump_dtx_option_parse(ctx, &info->dci_cmd_option.dci_dump_dtx, parsed);
+	}
+	if (same(cmd, COMMAND_NAME_CLEAR_DTX)) {
+		info->dci_cmd = DDB_CMD_CLEAR_DTX;
+		return clear_dtx_option_parse(ctx, &info->dci_cmd_option.dci_clear_dtx, parsed);
+	}
 
 	return -DER_INVAL;
 }
+
+int
+ddb_run_help(struct ddb_ctx *ctx)
+{
+	ddb_print(ctx, "Usage:\n");
+	ddb_print(ctx, "ddb [OPTIONS] <command>\n");
+	ddb_print(ctx, "\n");
+	ddb_print(ctx, "ddb (DAOS Debug) is a tool for connecting to a VOS file for "
+		       "the purpose of investigating and resolving issues.\n");
+
+	ddb_print(ctx, "\nOptions:\n");
+	ddb_print(ctx, "   -R, --run_cmd\n");
+
+	ddb_print(ctx, "\nCommands:\n");
+	ddb_print(ctx, "   help\t\t\tShow this help message\n");
+	ddb_print(ctx, "   quit\t\t\tExit interactive mode\n");
+	ddb_print(ctx, "   ls\t\t\tList the containers, objects, dkeys, akeys, "
+		       "or value depending\n");
+	ddb_print(ctx, "   dump_superblock\tDump the pool superblock information\n");
+	ddb_print(ctx, "   dump_value\t\tDump a value to a file\n");
+	ddb_print(ctx, "   rm\t\t\tRemove a branch of the VOS tree\n");
+	ddb_print(ctx, "   load\t\t\tLoad an updated or new value\n");
+	ddb_print(ctx, "   dump_ilog\t\tDump the ilog\n");
+	ddb_print(ctx, "   process_ilog\t\tProcess the ilog\n");
+	ddb_print(ctx, "   rm_ilog\t\tRemove all the ilog entries\n");
```

Comment:
I'm not sure it's entirely useful to remove _all_ ilog entries.  You probably just want to delete the key/object at that point.   Instead, I could potentially see some utillity in being able to modify or remove an ilog entry.

Even removing individual entries is tricky because you would also want to remove any associated updates.   For example, if you have an ilog entry for creating at epoch 5 and you remove it, you've orphaned any value written at or after 5.    Anyway, it's probably worth having a brainstorm session on this.  Is there a CR working group?

Reply:
@jolivier23 Thanks for the input. There is a CR working group ... I'll send you the invite.